[21기 박준우] Add: 회원가입 페이지 카카오 로그인 백엔드 연동 완료

AutumnWithJay opened the pull request:
## 수정 사항 간략한 한줄 요약
- Add: 카카오 로그인 백엔드 연동 완료
		 
## 수정 사항들 자세한 내용
- 카카오 간편 로그인을 완료하였습니다
- 백엔드와의 연동 역시 성공하여 토큰값과 유져 정보를 주고 다시 백엔드로부터 토큰값을 받았습니다(Handshaking?)

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 없습니다

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.
- [x] Git rebase와 squash를 했고 커밋 수가 하나 인것을 확인 했습니다.

dipsiiiiiiiiii commented on src/pages/Register/KakaoLogin.js:
```diff
@@ -0,0 +1,46 @@
+const KakaoLogin = history => {
```
함수이름, 파일명 둘 다 camelCase로 바꿔주세요!

AutumnWithJay replied:
수정했습니다 😁

dipsiiiiiiiiii commented on src/pages/Register/KakaoLogin.js:
```diff
@@ -0,0 +1,46 @@
+          ],
+        },
+        success: res => {
+          fetch('http://10.58.5.78:8000/users/signin', {
```
서버 주소는 계속 바뀔 수 있어서 한 곳에서 관리해 주세요!

AutumnWithJay replied:
수정했습니다 😄

dipsiiiiiiiiii commented on src/pages/Register/KakaoLogin.js:
```diff
@@ -0,0 +1,46 @@
+            .then(res => {
+              if (res.token) {
+                localStorage.setItem('Token', res.token);
+                console.log('로그인 되었습니다.');
+                history.push('/');
+              } else {
+                console.log('다시 확인해 주세요.');
```
확인용 콘솔이시면 지워 주시고 아니면 alert로 바꿔 주세요!

AutumnWithJay replied:
수정했습니다 😁

dipsiiiiiiiiii reviewed (commented):
카카오 로그인까지! 고생 많으셨습니다

AutumnWithJay committed:
Fix: 피드백 사항 반영완료
- Add: 카카오톡 로그인 버튼을 이미지로 대체
- Add: 문구 수정
- Fix: 피드백 수정사항 반영완료
- Fix: 로그인 환경변수 설정
- Fix: index.html 빈값 수정 완료

dipsiiiiiiiiii commented on public/index.html:
```diff
@@ -25,6 +25,12 @@
+  <script src="https://developers.kakao.com/sdk/js/kakao.js"></script>
+  <script>
+    // SDK를 초기화 합니다. 사용할 앱의 JavaScript 키를 설정해 주세요.
+    Kakao.init('68b2bd591aad95a05be5b92e54978afc');
```
키는 env로 분리해 주세요!

dipsiiiiiiiiii commented on public/index.html:
```diff
@@ -25,6 +25,12 @@
+  <script src="https://developers.kakao.com/sdk/js/kakao.js"></script>
+  <script>
+    // SDK를 초기화 합니다. 사용할 앱의 JavaScript 키를 설정해 주세요.
+    Kakao.init('68b2bd591aad95a05be5b92e54978afc');
```
이런 키 값들은 노출되지 않게 env로 분리해 주세요

AutumnWithJay replied:
수정했습니다 😄

dipsiiiiiiiiii commented on src/config.js:
```diff
@@ -1 +1,2 @@
 export const API = 'http://10.58.1.226:8000';
+export const loginAPI = 'http://10.58.5.78:8000';
```
이런 상수 값들은 컨벤션이 ```LOGIN_API``` 이렇게 해주시는게 좋습니다

AutumnWithJay replied:
수정했습니다~

dipsiiiiiiiiii reviewed (commented):
리뷰 반영해 주세요!
